File: Framework/src/models/NFRDefinition.java
package models;

public class NFRDefinition extends NFRAbsModel {

    //criar duas classes, uma pra segurança e outra pra desempenho
    //mudar os thresholdsbalance de cada um
    //enviar os dados do acelerometro de maneira criptografada para o servidor e esperar um ack do servidor para
    //mudar a cor da aplicação

    private String name;
    private int index;

    public void starNFRExecution(){

    }

    public String getName(){
        return this.name;
    }

    public int getIndex(){
        return this.index;
    }

    public void setName(String name){
        this.name = name;
    }

    public void setIndex(int index){
        this.index = index;
    }


}

File: Dart/Framework/models/NFRDefinition.dart
import 'NFRAbsModel.dart';

class NFRDefinition implements NFRAbsModel {
  late String _name;
  late int _index;

  void starNFRExecution() {}

  String getName() {
    return this._name;
  }

  int getIndex() {
    return this._index;
  }

  void setName(String name) {
    this._name = name;
  }

  void setIndex(int index) {
    this._index = index;
  }

  @override
  void notifyMonitor() {
    // TODO: implement notifyMonitor
  }
}
